### PR TITLE
Plotting in simple regression dialogue

### PR DESCRIPTION
**Pull request by stevekogo**
Added prediction interval plot to simple regression dialogue

### CODE REVIEW

**Comment by stevekogo** [comment]
@dannyparsons @bethanclarke I have implemented prediction interval plot in simple regression but I still need clarifications on two things.
-where will I get the value for alpha? Should I add a numeric up-down for this in the sub-dialogue? Or should I make the one for confidence interval work for both?
-should we be changing the show legend to either true or false? As of now, I have hard-coded it to false but I can add new checkbox to cater for this.

**Comment by bethanclarke** [comment]
Hi @stevekogo, all of these are great questions and sorry I didn't clarify any of these yesterday. alpha is a parameter that controls the transparency of the interval, so that can be left how it is. I will follow up with an email to answer your other questions,

File: instat/sdgSimpleRegOptions.vb [hunk 1]
```diff
@@ -92,10 +92,30 @@ Public Class sdgSimpleRegOptions
...
+    Private Sub FitPrediction()
+        FittedModel()
+        clsRPredFunction.SetRCommand("predict")
+        clsRDFFunction.SetRCommand("data.frame")
+        clsRPredFunction.AddParameter("object", clsRFunctionParameter:=dlgRegressionSimple.ucrBase.clsRsyntax.clsBaseFunction)
+        clsRPredFunction.AddParameter("interval", Chr(34) & "prediction" & Chr(34))
+        clsRDFFunction.AddParameter("", clsRFunctionParameter:=clsRPredFunction)
+        clsRaes_ribbon.SetRCommand("aes")
+        clsRaes_ribbon.AddParameter("ymin", clsRDFFunction.ToScript() & "$lwr")
+        clsRaes_ribbon.AddParameter("ymax", clsRDFFunction.ToScript() & "$upr")
```
Comment (dannyparsons) [inline instat/sdgSimpleRegOptions.vb]: This is great. I discussed "temporary columns" with David which is  what we get from predict i.e. columns needed for a particular analysis but we probably don't want them visible as part of the main data frame. Another example of "temporary columns" is lag variables in climatic analysis.
So we think these columns should be part of the data frame, and there is an extra piece of meta data to say which columns are "temporary" so we can do things like hide temporary columns by default etc. Until that's in place they will just be added like normal columns.
So now the output of predict just assign to new columns and then the ggplot is done on the original data still. I hope that makes sense?

**Comment by stevekogo** [comment]
@dannyparsons Please check the implementation of what you asked me on the simple regression dialogue.

**Comment by dannyparsons** [comment]
Great that this is working very well. I want to check the implementation with David.  I think what you've done is very sensible to get it working, I want to check with David if we should be modifying the RSyntax so that this can be done in other ways.

**Pull request closed without merging**